rust/src/container.rs

```diff
@@ -190,6 +193,7 @@ fn build_fs_mapping_recurse(
     path: &mut Utf8PathBuf,
     dir: &gio::File,
     state: &mut MappingBuilder,
+    parent_component: Option<String>,
```

Comment:
I think this should be `Option<&str>` - no reason to require an owned value AFAICS